backend/app/routers/experiments.py

```diff
@@ -139,34 +123,38 @@ async def update_experiment(
         )
 
     await Experiment.replace(experiment_to_save)
-    return experiment_to_save.map_to_response()
+    return experiment_to_save.map_to_response(user)
 
 
 @router.delete("/experiments/{id}", response_model=None)
 async def delete_experiment(
-    id: PydanticObjectId, user: dict = Depends(get_current_user(required=True))
+    id: PydanticObjectId,
+    user: dict = Depends(get_current_user(required=True)),
+    workflow_engine: WorkflowEngineBase = Depends(ReanaService.get_service),
 ) -> Any:
     await get_experiment_if_accessible_or_raise(id, user, write_access=True)
-    exist_runs = await ExperimentRun.find(ExperimentRun.experiment_id == id).count() > 0
-    if exist_runs:
-        raise HTTPException(
-            status_code=status.HTTP_403_FORBIDDEN,
-            detail="This experiment cannot be deleted",
-        )
+    runs = await ExperimentRun.find(ExperimentRun.experiment_id == id).to_list()
+
+    for run in runs:
+        await workflow_engine.delete_workflow(run)
```

Comment:
Can we add a method `delete_run()` and reuse it in here as well as in the endpoint for deleting a run?

Reply:
I meant a separate ExperimentRun method for removing everything related to run.
```python
await workflow_engine.delete_workflow(run)
await run.delete_files()
await ExperimentRun.find(ExperimentRun.id == run.id).delete()
```